gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -924,6 +926,227 @@ public void deleteObjects(List<StorageResourceId> fullObjectNames) throws IOExce
+              folders.size(),
+              storageOptions.getBatchThreads(),
+              "batchFolderDelete");
+
+      // Map to store number of children for each parent object
+      HashMap<String, Long> occurenceOfFolder = new HashMap<>();
+      for (FolderInfo currentFolder : folders) {
+        // inserting each folder
+        if (!occurenceOfFolder.containsKey(currentFolder)) {
+          occurenceOfFolder.put(currentFolder.getFolderName(), 0L);
+        }
+
+        String parentFolder = currentFolder.getParentFolderName();
+
+        if (parentFolder != "" && parentFolder != null) {
+          occurenceOfFolder.put(
+              parentFolder,
+              (occurenceOfFolder.containsKey(parentFolder)
+                      ? occurenceOfFolder.get(parentFolder)
+                      : 0)
+                  + 1);
+        }
+      }
+
+      BlockingQueue<FolderInfo> folderDeleteBlockingQueue = new LinkedBlockingQueue<>();
+      for (FolderInfo currentFolder : folders) {
+        if (occurenceOfFolder.get(currentFolder.getFolderName()) == 0L) {
+          addFolderResourceInBlockingQueue(
+              currentFolder, folderDeleteBlockingQueue, innerExceptions);
+        }
+      }
+
+      while (!occurenceOfFolder.isEmpty() && innerExceptions.isEmpty()) {
+        while (!folderDeleteBlockingQueue.isEmpty()) {
+          FolderInfo folderToDelete = null;
+          try {
+            folderToDelete = folderDeleteBlockingQueue.poll(1, TimeUnit.MINUTES);
+            queueSingleFolderDelete(
+                folderToDelete,
+                occurenceOfFolder,
+                folderDeleteBlockingQueue,
+                innerExceptions,
+                batchHelper,
+                1);
+
+          } catch (InterruptedException e) {
+            logger.atInfo().log("Blocking queue is not receiving any more folder resource");
+            break;
+          }
+        }
+      }
+
+      // deleting any remaining resources
+      batchHelper.flush();
+    }
+
+    if (!innerExceptions.isEmpty()) {
+      GoogleCloudStorageEventBus.postOnException();
+      throw GoogleCloudStorageExceptions.createCompositeException(innerExceptions);
+    }
+  }
+
+  /** Helper function to delete a single folder resource */
+  public void queueSingleFolderDelete(
+      final FolderInfo folder,
+      HashMap<String, Long> occurenceOfFolder,
+      BlockingQueue<FolderInfo> folderDeleteBlockingQueue,
+      final KeySetView<IOException, Boolean> innerExceptions,
+      final BatchHelper batchHelper,
+      final int attempt)
+      throws IOException {
+    checkArgument(
+        folder.getBucket() != "" && folder.getBucket() != null,
+        String.format("No bucket for folder resource %s", folder.toString()));
+
+    checkArgument(
+        folder.getFolderName() != "" && folder.getFolderName() != null,
+        String.format("No folder path for folder resource %s", folder.toString()));
+
+    final String bucketName = folder.getBucket();
```

Comment:
Move this up and then you can use this in the checkArgument

Reply:
Updated